src/NLog/Targets/FileTarget.cs

```diff
@@ -1908,6 +1915,24 @@ private bool PreviousLogOverlappedPeriod(LogEventInfo logEvent, DateTime lastWri
         }
 
         /// <summary>
+        /// Calculate the DateTime of the requested day of the week.
+        /// </summary>
+        /// <param name="previousLogEventTimestamp">The DateTime of the previous log event.</param>
+        /// <param name="dayOfWeek">The next occuring day of the week to return a DateTime for.</param>
+        /// <returns>The DateTime of the next occuring dayOfWeek.</returns>
+        /// <remarks>For example: if previousLogEventTimestamp is Thursday 2017-03-02 and dayOfWeek is Sunday, this will return
+        ///  Sunday 2017-03-05. If dayOfWeek is Thursday, this will return *next* Thursday 2017-03-09.</remarks>
+        public DateTime CalculateNextWeekday(DateTime? previousLogEventTimestamp, DayOfWeek dayOfWeek)
+        {
+            // Shamelessly taken from http://stackoverflow.com/a/7611480/1354930
+            int start = (int)previousLogEventTimestamp.Value.DayOfWeek;
```

Comment:
This function is only called by `PreviouisLogOverlappedPeriod()`. At the begining of that function, a check for nullref with `if(!previousLogEventTimestamp.HasValue) return false;` already exists.

How should I adding nullref handling to `CalcualteNextWeekday`?
1. assume `previousLogEventTimestamp = DateTime.Now` and return the next `dayOfWeek`?
2. simply return `DateTime.Now`?
3. throw the error up the chain?
4. alternatively, I can edit `PreviousLogOverlappedPeriod()` to convert the nullable `previousLogEventTimestamp` to a non-nullable type and then not allow nulls in `CalculateNextWeekday`

I don't know enough about the codebase to say if 1 or 2 would cause other issues (such as never archiving), so I'm leaning towards option 4. Thoughts?